ui/panels-plugin/src/plugins/stat-chart/StatChartPanel.tsx

```diff
@@ -27,15 +31,17 @@ export function StatChartPanel(props: StatChartPanelProps) {
     contentDimensions,
   } = props;
 
-  const { queryResults, isLoading } = useDataQueries();
-  const chartData = useChartData(queryResults[0]?.data, calculation);
+  const { queryResults, isLoading, isFetching } = useDataQueries();
+  const statChartData = useStatChartData(queryResults, calculation);
+  const isMultiSeries = statChartData.length > 1;
+
   const chartsTheme = useChartsTheme();
 
   if (queryResults[0]?.error) throw queryResults[0]?.error;
 
   if (contentDimensions === undefined) return null;
 
-  if (isLoading === true) {
+  if (isLoading || isFetching) {
```

Comment:
Nit: you should only need `isFetching` since it encapsulates `isLoading`. `isLoading` is only the first time